corehq/apps/user_importer/validation.py

```diff
@@ -382,3 +389,52 @@ def validate_spec(self, spec):
                 if error_values:
                     errors_formatted = ' and '.join(error_values)
                     return self.error_existing_user.format(self.confirmation_sms_header, errors_formatted)
+
+
+class LocationAccessValidator(ImportValidator):
+    error_message_user_access = _("Based on your locations do not have permission to edit this user or user "
+                                  "invitation")
+    error_message_location_access = _("You do not have permission to assign or remove these locations: {}")
+
+    def __init__(self, domain, upload_user, location_cache, is_web_user_import):
+        super().__init__(domain)
+        self.upload_user = upload_user
+        self.location_cache = location_cache
+        self.is_web_user_import = is_web_user_import
+
+    def validate_spec(self, spec):
+        from corehq.apps.user_importer.importer import find_location_id
+        # 1. Get current locations for user or user invitation and ensure user can edit it
+        username = spec.get('username')
+        current_locs = []
+        if self.is_web_user_import:
+            editable_user = CouchUser.get_by_username(username, strict=True)
+            if not editable_user:
```

Comment:
This would check non-Existing Users being added to the domain. But I think it would miss existing users that don't belong to domain right? I think the permission check for Invitation should be done regardless if `editable_user` exists

Reply:
that looks good to me!